tests/engines/collections/integration/components/collection-submission-confirmation-modal/component-test.ts

```diff
@@ -1,103 +1,184 @@
 import { click, render } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
-import { t } from 'ember-intl/test-support';
+import { t, TestContext } from 'ember-intl/test-support';
 import { setupEngineRenderingTest } from 'ember-osf-web/tests/helpers/engines';
 import { module, test } from 'qunit';
 
 module('Integration | Component | collection-submission-confirmation-modal', hooks => {
     setupEngineRenderingTest(hooks, 'collections');
+    let externalSaveClicked = false;
+    let externalCancelClicked = false;
+    let externalResubmitClicked = false;
 
-    test('it renders', async function(assert) {
-        this.set('noop', () => { /* noop */ });
-        await render(hbs`
-        {{collection-submission-confirmation-modal
-            resubmitToCollection=(action this.noop)
-            openModal=true addToCollection=(action this.noop)
-            cancel=(action this.noop)
-        }}`);
-        assert.dom('[data-test-collection-submission-confirmation-modal-header]').hasText(
-            t('collections.collection_submission_confirmation_modal.header').toString(),
-        );
-        assert.dom('[data-test-collection-submission-confirmation-modal-body]').hasText(
-            t('collections.collection_submission_confirmation_modal.body').toString(),
+    hooks.beforeEach(async function(this: TestContext) {
+        externalSaveClicked = false;
+        externalCancelClicked = false;
+        externalResubmitClicked = false;
+
+        this.set('externalResubmitAction', () => {
+            externalResubmitClicked = true;
+        });
+        this.set('externalSaveAction', () => {
+            externalSaveClicked = true;
+        });
+        this.set('externalCancelAction', () => {
+            externalCancelClicked = true;
+        });
+    });
+
+    test('it renders', async function(this: TestContext, assert) {
+        // Given the component is rendered
+        await render(
+            hbs`<CollectionsSubmission::CollectionSubmissionConfirmationModal
+                @resubmitToCollection={{ this.externalResubmitAction }}
+                @openModal=true
+                @addToCollection={{ this.externalSaveAction}}
+                @cancel={{ this.externalCancelAction }}
+            />`,
         );
+        // Then I assert the modal header text
+        assert.dom('[data-test-collection-submission-confirmation-modal-header]')
+            .hasText('Submit project', 'The modal header is wrong');
```

Comment:
Is the modal header supposed to be wrong when this test passes, or did you write the bdd comment for the failure state?